Code/Framework/AzToolsFramework/Tests/PythonLoaderTests.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/Entity.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/sort.h>
+#include <AzTest/AzTest.h>
+#include <AzTest/Utils.h>
+#include <AzToolsFramework/Application/ToolsApplication.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <AzToolsFramework/Entity/EditorEntityContextComponent.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzFramework/IO/LocalFileIO.h>
+#include <AzToolsFramework/API/PythonLoader.h>
+
+namespace UnitTest
+{
+    class AzToolsFrameworkPythonLoaderFixture
+        : public ToolsApplicationFixture<>
+    {
+    protected:
+        AZ::Test::ScopedAutoTempDirectory m_tempDirectory;
+
+        static constexpr const char* s_testEnginePath = "O3de_path";
+        static constexpr const char* s_testEnginePathHashId = "1af80774";
```

Comment:
FYI: I made the Sha1 hash functions constexpr, so they can be called at compile time to pre-calculate the hash based on the engine path.

That way instead of a magic value of `1af80774` needing to be hard coded in the source file, an immediately invoked lambda can be used to calculate the hash based on the `s_testEnginePath` constant.

```suggestion
        static constexpr AZStd::string_view  s_testEnginePath = "O3de_path";        
        static constexpr AZ::u32 s_testEnginePathHashId = []() -> AZ::u32
        {
            AZ::Sha1 pathSha1;
            pathSha1.ProcessBytes(AZStd::as_bytes(AZStd::span(s_testEnginePath)));
            AZ::Sha1::DigestType digestType{};
            pathSha1.GetDigest(digestType);
            return digestType[0];
        }();
```

Reply:
A couple of errors from that suggestion:
```
1>D:\github\o3de\Code\Framework\AzToolsFramework\Tests\PythonLoaderTests.cpp(40,44): error C2440: 'initializing': cannot convert from 'initializer list' to 'AZ::Sha1::DigestType'
1>            AZ::Sha1::DigestType digestType{};
```
And if I change it to `AZ::u32 digestType[5];`, it progresses to the next error
```
error C2131: expression did not evaluate to a constant
```
So i'll leave it as is, there is no reason to update the engine path anyways.